app/src/js/server/create_project.ts

```diff
@@ -79,35 +79,72 @@ export async function parseForm (
  * Parses item, category, and attribute files from paths
  */
 export async function parseFiles (
-  labelType: string, files: { [key: string]: string }, itemsRequired: boolean)
+  labelType: string, files: { [key: string]: string },
+  storage: Storage, itemsRequired: boolean)
   : Promise<FormFileData> {
-  return Promise.all([
-    parseItems(files, itemsRequired),
-    parseSensors(files),
-    parseTemplates(files),
-    parseAttributes(files, labelType),
-    parseCategories(files, labelType)])
+  const items = parseItems(files, itemsRequired, storage)
+
+  const categories: Promise<Category[]> = readFileWithDefault(
+    files, FormField.CATEGORIES, getDefaultCategories(labelType), storage)
```

Comment:
Can we do `files[FormField.CATEGORIES]` to simplify the interface of `readFileWithDefault`?

Reply:
There might be a key error if there is no category file. In that case, the function uses the default.

Reply:
In that case, it is clearer to use `_.get(files, FormField.CATEGORIES)`. It will return `undefined` if the key doesn't exist. Then the type for the file path can be `string | undefined`.

Reply:
Fixed